### DISCUSSION TITLE
Default VS Code extensions?

### DISCUSSION THREAD

**Discussion by kcpevey** [nebari-dev/nebari#1775] [Q&A]
I'd like to have the Python extension for VS Code installed by default for all users. Is this possible?

**Comment by pavithraes** [comment]
Sharing notes from Slack:

> Tania: It should be [installed]

> Dharhas: we also need to fix the vscode git extension. I had an issue opened for that a month or so ago.

> Adam: I'm not aware of this capability at the moment.  I'm not sure how hard it would be to add either.  I think the vs-code integration is called code-server https://github.com/coder/code-server so searching in Nebari and/or code-server docs might help produce more info.  Off the top of my head, I think jupyterhub allows a pre and post start up script that runs when the user pod starts up and that might be useful to do that, but that's about all I know at the moment.

**Reply by mcg1969** [reply]
+1 on this feature. I would recommend Python, Jupyter, perhaps an R extension, and RedHat's YAML extension.

**Reply by mcg1969** [reply]
I've worked with code-server before. It is relatively straightforward to include extensions in the default install.

**Comment by dharhas** [comment]
https://gist.github.com/mcg1969/f45772ada77765f57891e2907175b530
